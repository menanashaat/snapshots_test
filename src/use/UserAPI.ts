// api.ts
import axios from 'axios'

// Function to fetch users from the API
export const getUsers = async () => {
  const response = await axios.get('https://61cb0d6d194ffe0017788b36.mockapi.io/Users')
  return response.data // Return the fetched data
}
