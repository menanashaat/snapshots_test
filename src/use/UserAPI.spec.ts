import { describe, it, expect, beforeAll } from 'vitest'
import { getUsers } from './UserAPI' // Adjust the path as necessary

// Helper function to extract property names
const extractKeys = (users: Record<string, any>[]) => {
  return users.map((user) => Object.keys(user).sort()) // Sort keys to ensure consistent ordering
}

describe('User API Snapshot Test', () => {
  let users: any[] // Type can be defined as needed, e.g., User[]

  beforeAll(async () => {
    users = await getUsers() // Fetch users before running tests
  })

  it('matches the snapshot for properties', () => {
    const userKeys = extractKeys(users)

    // Check that each user has the required properties
    userKeys.forEach((keys) => {
      expect(keys).toEqual(
        expect.arrayContaining(['FristName', 'LastName', 'Email', 'id', 'Phone'])
      )
    })
    // Take a snapshot of the keys
    expect(userKeys).toMatchSnapshot()
  })
})
