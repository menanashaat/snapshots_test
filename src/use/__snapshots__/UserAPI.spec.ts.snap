// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`User API Snapshot Test > matches the snapshot for properties 1`] = `
[
  [
    "Email",
    "FristName",
    "LastName",
    "Phone",
    "id",
  ],
  [
    "Email",
    "FristName",
    "LastName",
    "Phone",
    "id",
  ],
  [
    "Email",
    "FristName",
    "LastName",
    "Phone",
    "id",
  ],
  [
    "Email",
    "FristName",
    "LastName",
    "Phone",
    "id",
  ],
  [
    "Email",
    "FristName",
    "LastName",
    "Phone",
    "id",
  ],
  [
    "Email",
    "FristName",
    "LastName",
    "Phone",
    "id",
  ],
  [
    "Email",
    "FristName",
    "LastName",
    "Phone",
    "id",
  ],
]
`;
